apps/hexagon_dma/process_nv12.cpp

```diff
@@ -47,11 +47,19 @@ int main(int argc, char **argv) {
     // The Last parameter 0 indicate DMA Read
     halide_hexagon_dma_prepare_for_copy_to_host(nullptr, input, dma_engine, false, eDmaFmt_NV12);
 
-    Halide::Runtime::Buffer<uint8_t> output(width, height, 2);
+    //Halide::Runtime::Buffer<uint8_t> output(width, height, 2);
+    Halide::Runtime::Buffer<uint8_t> output(width, (3 * height) / 2);
+    output.embed(2, 0);
+    output.raw_buffer()->dim[1].extent = height;
+    output.raw_buffer()->dim[2].extent = 2;
+    output.raw_buffer()->dim[2].stride = width * height;
+      
     Halide::Runtime::Buffer<uint8_t> output_y = output.cropped(2, 0, 1);    // Luma plane only
```

Comment:
This also doesn't quite look right. I think the whole buffer construction should look like this:

    Halide::Runtime::Buffer<uint8_t> output(width, (3 * height) / 2);

    // Extract the Y plane
    Halide::Runtime::Buffer<uint8_t> output_y = output.cropped(1, 0, height);

    // Extract the UV plane.
    Halide::Runtime::Buffer<uint8_t> output_uv = output.cropped(1, height, height/2);

    // Add a channels dimension. At this point, the buffer has dimensions width x height/2 x 1
    output_uv.embed(2, 0);

    // Rewrite the x dimension so the buffer now has dimensions width/2 x height/2 x 1, though the x dimension has stride 2
    output_uv.raw_buffer()->dim[0].extent = width / 2;
    output_uv.raw_buffer()->dim[0].stride = 2;

    // Verify that the height is what we expect.
    assert(output_uv.raw_buffer()->dim[1].extent == height / 2);

    // Make the channels dimension have extent 2, and stride 1 (pairs of UV values are stored adjacently in memory).
    output_uv.raw_buffer()->dim[2].stride = 1; 
    output_uv.raw_buffer()->dim[2].extent = 2;  
    
    // Now the buffer has dimensions width/2 x height/2 x 2, and the channels are stored interleaved.

Reply:
Is the only change from what is committed to this branch the pipeline you copy/pasted into this comment?

Reply:
It looks like the host code is still checking for a planar chroma buffer, but the pipeline you showed above outputs an interleaved buffer, so that seems like the source of a potential problem (though that should cause an assert to fail, not bad output).

For reference, I'm referring to line 75 of apps/hexagon_dma/process_nv12.cpp, which appears to be loading a plane (the stride of 'c' is width*height) from the reference input.